sdk/storage/AzureStorageBlob/Source/Operation/BlockOperations.swift

```diff
@@ -67,11 +68,15 @@ internal class BlockOperation: ResumableOperation {
                 options: downloader.options
             )
             chunkDownloader.download { result, httpResponse in
-                if self.isCancelled || self.isPaused { return }
+                if !transfer.isActive {
+                    group.leave()
```

Comment:
I believe this was the root cause of the issue. All code paths after a `group.enter()` call must have a matching `group.leave()` or the lifecycle for the operation hangs, starving the queue of threads.